packages/sandcastle/src/App.tsx

```diff
@@ -241,11 +247,46 @@ function App() {
           committedCode: action.code ?? state.code,
           committedHtml: action.html ?? state.html,
           runNumber: state.runNumber + 1,
+          dirty: false,
+        };
+      }
+      case "resetDirty": {
+        return {
+          ...state,
+          dirty: false,
         };
       }
     }
   }, initialState);
 
+  useEffect(() => {
+    const host = window.location.host;
+    let envString = "";
+    if (host.includes("localhost") && host !== "localhost:8080") {
+      // this helps differentiate tabs for local sandcastle development or other testing
+      envString = `${host.replace("localhost:", "")} `;
+    }
+
+    const baseTitle = "Cesium Sandcastle";
+    const dirtyIndicator = codeState.dirty ? "*" : "";
+    if (title !== "") {
+      document.title = `${envString}${title}${dirtyIndicator} | ${baseTitle}`;
+    } else {
+      document.title = `${envString}${baseTitle}`;
+    }
```

Comment:
1. Why shouldn't we show the dirty indicator on named sandcastles?
2. Assuming we can, couldn't we simplify this to thew following?

```suggestion
      document.title = `${envString}${title}${dirtyIndicator} | ${baseTitle}`;
```

Reply:
This check makes sure that we only add the title and the ` | ` to the page title when there's actually a title to prepend in front of the bar. The page starts with an empty title `""` in state when it initially loads because we don't know what it will be yet.
If a specific gallery demo is not loaded then we replace it with "New Sandcastle" thus providing a title and it _will_ show the dirty indicator.